rest_api/src/main/java/tqs/cloudit/services/AuthenticationService.java

```diff
@@ -24,30 +23,49 @@
  */
 @Service
 public class AuthenticationService {
+    
     @Autowired
     private UserRepository userRepository;
     
     @Autowired
     private AreaRepository areaRepository;
     
-    private BCryptPasswordEncoder bcpe=new BCryptPasswordEncoder();
+    private BCryptPasswordEncoder bcpe = new BCryptPasswordEncoder();
     private Logger l = LoggerFactory.getLogger(AuthenticationService.class);
     
     public ResponseEntity register(User user) {
+        if(!user.allDefined()){
+            JSONObject response = new JSONObject();
+            response.put("status", 1);
+            response.put("message", "Registration invalid. When registering you need to provide username, password, name, email, type of user and optionally interest areas.");
+            return ResponseEntity.badRequest().body(response);
+        }
+        if(userRepository.getUsernames().contains(user.getUsername())){
+            JSONObject response = new JSONObject();
+            response.put("status", 1);
+            response.put("message", "Registration invalid. Username must be unique. This username already exists.");
+            return ResponseEntity.badRequest().body(response);
+        }
+        if(userRepository.getMails().contains(user.getEmail())){
+            JSONObject response = new JSONObject();
+            response.put("status", 1);
+            response.put("message", "Registration invalid. Your email must be unique. This email is already registered in the platform.");
+            return ResponseEntity.badRequest().body(response);
+        }
+        
         tqs.cloudit.domain.persistance.User pUser=new tqs.cloudit.domain.persistance.User(user);
-        for(Area a : pUser.getInterestedAreas()){
-            if(this.areaRepository.existsByArea(a.getArea())==0)
-                this.areaRepository.save(a);
+        if(pUser.getInterestedAreas()!=null){
+            for(Area a : pUser.getInterestedAreas()){
+                if(this.areaRepository.existsByArea(a.getArea())==0)
+                    this.areaRepository.save(a);
+            }
         }
         pUser.setPassword(this.bcpe.encode(pUser.getPassword()));
         this.userRepository.save(pUser);
-        return ResponseEntity.ok("firmeza");
-    }
-    
-    public ResponseEntity login(Credentials cred) {
-        if(this.bcpe.matches(cred.getPassword(),this.userRepository.getPass(cred.getUsername()))){
-            return ResponseEntity.ok("firmeza");
-        }
-        return ResponseEntity.ok("não firmeza");
+        
+        JSONObject response = new JSONObject();
+        response.put("status", 0);
```

Comment:
I think we should return the status of the response through HTTP status codes, what maybe needs to be discussed between the group.
```Java
        ResponseEntity(HttpStatus.Accepted, "Registered with success");
```